### ISSUE TITLE
Redesign core implementation

### ISSUE THREAD

**Issue by vlad107** [aelve/codesearch#71]
- [x] Not use Index methods from outside world
- [x] Merge [Language]Index and [Langauge]Sources into single object [Language]
   (or remember why this separation makes sense)
- [x] Similar functions should be implemented only once in a base class:
    - [x] csearch
    - [x] ContentByURI
- [x] Define SearchArguments class containing 
    * searchQuery: String
    * insensitiveFlag: Boolean
    * preciseMatchFlag: Boolean
    * sourcesOnlyFlag: Boolean
- [ ] Investigate more into implementation design 
- [x] move execution search to execution context
- [ ] add enum Language with all information (e.g. extensions) - replace util.Helper

**Comment by neongreen** [comment]
I'm going to close this one. I'm not sure what `add enum Language with all information (e.g. extensions) - replace util.Helper` means, but a new issue can be created for this.
